api/app/src/providers/withings.ts

```diff
@@ -99,7 +99,12 @@ export class Withings extends Provider implements OAuth2 {
 
   async checkRefreshToken(token: string, connectedUser: ConnectedUser): Promise<Token> {
     const access_token = JSON.parse(token);
-    const isExpired = access_token.expires_at * 1000 - (Date.now() + 120 * 1000) <= 0;
+    const bufferSeconds = 600;
+    const isExpired =
+      convert(access_token.expires_at).from("s").to("ms") -
+        Date.now() +
+        convert(bufferSeconds).from("s").to("ms") <=
+      0;
```

Comment:
We have this in two places, maybe worth exposing as a function and calling it from those places?